feat(lts): add new resource to support cross account access (#4423)
huaweicloud/terraform-provider-huaweicloud: docs/resources/lts_cross_account_access.md

---
subcategory: "Log Tank Service (LTS)"
---

# huaweicloud_lts_cross_account_access

Manages an LTS cross account access resource within HuaweiCloud.

-> **NOTE:** 1. An agency relationship has been created.
<br/> 2. Before data synchronization is complete, data in the target and source log streams may be different.
         Check back later in one hour.
<br/> 3. After you configure cross-account access, if account A deletes the agency from IAM, LTS cannot detect the
         deletion and the cross-account ingestion still takes effect. If the cross-account access configuration is

>          no longer used, notify account B to delete it.

Reviewer: ```suggestion
-> **NOTE:** Somethings you should know before using this resource.
<br/> 1. Before creating this resource, you need to prepare an agency relationship.
<br/> 2. Before data synchronization is complete, data in the target and source log streams may be different.
         Check back later in one hour.
<br/> 3. After you configure cross-account access, if account A deletes the agency from IAM, LTS cannot detect the
         deletion and the cross-account ingestion still takes effect. If the cross-account access configuration is
         no longer used, notify account B to delete it.
```

Author: done

## Example Usage

```hcl
variable "name" {}
variable "agency_group_id" {}
variable "agency_stream_id" {}
variable "agency_group_name" {}
variable "agency_stream_name" {}
variable "log_group_id" {}
variable "log_stream_id" {}
variable "log_group_name" {}
variable "log_stream_name" {}
variable "agency_name" {}
variable "agency_domain_name" {}
variable "agency_project_id" {}

> resource "huaweicloud_lts_cross_account_access" "test" {
>   name                  = var.name
>   log_agencystream_name = var.agency_stream_name
>   log_agencystream_id   = var.agency_stream_id
>   log_agencygroup_name  = var.agency_group_name
>   log_agencygroup_id    = var.agency_group_id
>   log_stream_name       = var.log_stream_name
>   log_stream_id         = var.log_stream_id
>   log_group_name        = var.log_group_name
>   log_group_id          = var.log_group_id
>   agency_project_id     = var.agency_project_id
>   agency_domain_name    = var.agency_domain_name
>   agency_name           = var.agency_name
> }

Reviewer: ```suggestion
resource "huaweicloud_lts_cross_account_access" "test" {
  name                  = var.name
  agency_project_id     = var.agency_project_id
  agency_domain_name    = var.agency_domain_name
  agency_name           = var.agency_name

  # Relevant information of the delegated party.
  log_agencystream_name = var.agency_stream_name
  log_agencystream_id   = var.agency_stream_id
  log_agencygroup_name  = var.agency_group_name
  log_agencygroup_id    = var.agency_group_id
 
  # Relevant information of the delegatee party.
  log_stream_name       = var.log_stream_name
  log_stream_id         = var.log_stream_id
  log_group_name        = var.log_group_name
  log_group_id          = var.log_group_id
}
```
The necessary separation and comments help to understand the example.

Author: done

* `log_agencygroup_name` - (Required, String, ForceNew) Specify the log group name that already exists in the
  delegated account. Changing this creates a new resource.

* `log_agencystream_id` - (Required, String, ForceNew) Specifies the log stream ID that already exists in the
  delegated account. Changing this creates a new resource.

* `log_agencystream_name` - (Required, String, ForceNew) Specifies the log stream name that already exists in the
  delegated account. Changing this creates a new resource.

* `log_group_id` - (Required, String, ForceNew) Specify the log group ID that already exists in the
  delegatee account. Changing this creates a new resource.

* `log_group_name` - (Required, String, ForceNew) Specify the log group name that already exists in the
  delegatee account. Changing this creates a new resource.

* `log_stream_id` - (Required, String, ForceNew) Specifies the log stream ID that already exists in the
  delegatee account. Changing this creates a new resource.

* `log_stream_name` - (Required, String, ForceNew) Specifies the log stream name that already exists in the
  delegatee account. Changing this creates a new resource.

> * `tags` - (Optional, Map) Specifies the key/value to attach to the cross account access.

Reviewer: ```suggestion
* `tags` - (Optional, Map) Specifies the key/value pairs to associate with the cross account access.
```
Keep consistent with the description for other documents.

Author: done

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The resource ID.

* `access_config_type` - The log access configuration type.

> * `created_at` - The creation time of the cross account access.

Reviewer: what format is？
with timezone or not？

Author: done

  delegated account. Changing this creates a new resource.

* `log_agencystream_name` - (Required, String, ForceNew) Specifies the log stream name that already exists in the
  delegated account. Changing this creates a new resource.

* `log_group_id` - (Required, String, ForceNew) Specify the log group ID that already exists in the
  delegatee account. Changing this creates a new resource.

* `log_group_name` - (Required, String, ForceNew) Specify the log group name that already exists in the
  delegatee account. Changing this creates a new resource.

* `log_stream_id` - (Required, String, ForceNew) Specifies the log stream ID that already exists in the
  delegatee account. Changing this creates a new resource.

* `log_stream_name` - (Required, String, ForceNew) Specifies the log stream name that already exists in the
  delegatee account. Changing this creates a new resource.

* `tags` - (Optional, Map) Specifies the key/value pairs to associate with the cross account access.

## Attribute Reference

In addition to all arguments above, the following attributes are exported:

* `id` - The resource ID.

* `access_config_type` - The log access configuration type.

> * `created_at` - The creation time of the cross account access, in UTC format.

Reviewer: ```suggestion
* `created_at` - The creation time of the cross account access, in RFC3339 format.
```

Author: done
